dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/BaseClasses.mo

```diff
@@ -312,5 +343,67 @@ equation
+
+      parameter Types.PerUnit MdSat0PPu "Start value of direct axis saturated mutual inductance in p.u.";
+      parameter Types.PerUnit MqSat0PPu "Start value of quadrature axis saturated mutual inductance in p.u.";
+      parameter Types.PerUnit LambdaAirGap0Pu "Start value of total air gap flux in p.u.";
+      parameter Types.PerUnit LambdaAD0Pu "Start value of common flux of direct axis in p.u";
+      parameter Types.PerUnit LambdaAQ0Pu "Start value of common flux of quadrature axis in p.u";
+      parameter Types.PerUnit Mds0Pu "Start value of direct axis saturated mutual inductance in the case when the total air gap flux is aligned on the direct axis in p.u.";
+      parameter Types.PerUnit Mqs0Pu "Start value of quadrature axis saturated mutual inductance in the case when the total air gap flux is aligned on the quadrature axis in p.u.";
+      parameter Types.PerUnit Cos2Eta0 "Start value of the common flux of direct axis contribution to the total air gap flux in p.u.";
+      parameter Types.PerUnit Sin2Eta0 "Start value of the common flux of quadrature axis contribution to the total air gap flux in p.u.";
+      parameter Types.PerUnit Mi0Pu "Start value of intermerdiary axis saturated mutual inductance in p.u.";
+
+      // Saturated mutual inductances and related variables
+      Types.PerUnit MdSatPPu(start = MdSat0PPu) "Direct axis saturated mutual inductance in p.u.";
+      Types.PerUnit MqSatPPu(start = MqSat0PPu) "Quadrature axis saturated mutual inductance in p.u.";
+      Types.PerUnit lambdaAirGapPu(start = LambdaAirGap0Pu) "Total air gap flux in p.u.";
+      Types.PerUnit lambdaADPu(start = LambdaAD0Pu) "Common flux of direct axis in p.u.";
+      Types.PerUnit lambdaAQPu(start = LambdaAQ0Pu) "Common flux of quadrature axis in p.u.";
+      Types.PerUnit mdsPu(start = Mds0Pu) "Direct axis saturated mutual inductance in the case when the total air gap flux is aligned on the direct axis in p.u.";
+      Types.PerUnit mqsPu(start = Mqs0Pu) "Direct axis saturated mutual inductance in the case when the total air gap flux is aligned on the quadrature axis in p.u.";
+      Types.PerUnit sin2Eta(start = Sin2Eta0) "Common flux of quadrature axis contribution to the total air gap flux in p.u.";
+      Types.PerUnit cos2Eta(start = Cos2Eta0) "Common flux of direct axis contribution to the total air gap flux in p.u.";
+      Types.PerUnit miPu(start = Mi0Pu) "Intermerdiary axis saturated mutual inductance in p.u.";
+
+  equation
+
+    if running.value then
+
+      // Mutual inductances saturation
+      lambdaADPu = MdSatPPu*(idPu + ifPu + iDPu);
+      lambdaAQPu = MqSatPPu*(iqPu + iQ1Pu + iQ2Pu);
+      lambdaAirGapPu^2 = lambdaADPu^2 + lambdaAQPu^2;
+      mdsPu = MdPPu / (1 + md*lambdaAirGapPu^nd);
+      mqsPu = MqPPu / (1 + mq*lambdaAirGapPu^nq);
+      lambdaADPu^2 = cos2Eta * lambdaAirGapPu^2;
+      lambdaAQPu^2 = sin2Eta * lambdaAirGapPu^2;
```

Comment:
Any reason for using the squared value formulation rather than the root square one. I believe it leaves a degree of freedom on the variable sign.

Reply:
In the doc the equation was in the form Cos2Eta0 = LambdaAD0Pu^2 / LambdaAirGap0Pu^2 like in the init algorithm.
I think I wrote it differently at first to avoid potential numerical issues when LambdaAirGap0Pu=0, but now that we use start values I can revert it back its original formulation